### PR TITLE
playbook exp/gen Telemetry

### PR DESCRIPTION
**Pull request by goneri**
Jira Issue: <https://issues.redhat.com/browse/AAP-21502>

## Description

Schema1
-------

Accepts the following events:

- `playbookGenerationAction`
- `playbookExplanationFeedback`

Schema2
-------

Two new events:

- `AnalyticsPlaybookExplanationGenerated`
- `AnalyticsPlaybookGenerationWizard`

## Production deployment

- [x] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

**Comment by goneri** [comment]
The `pip-audit` issue is addressed by https://github.com/ansible/ansible-ai-connect-service/pull/1133

File: ansible_wisdom/ai/api/utils/analytics_telemetry_model.py [hunk 1]
```diff
@@ -23,6 +23,8 @@ class AnalyticsTelemetryEvents(Enum):
     RECOMMENDATION_GENERATED = "Recommendation Generated"
     RECOMMENDATION_ACTION = "Recommendation Action"
     PRODUCT_FEEDBACK = "Product Feedback"
+    PLAYBOOK_GENERATION_ACTION = "Action on the Playbook Generation Wizard"
+    PLAYBOOK_EXPLANATION_GENERATED = "Playbook Explanation Generated"
```
Comment (robinbobbitt) [inline ansible_wisdom/ai/api/utils/analytics_telemetry_model.py:27]: Per yesterday's meeting, we don't yet have a plan for analytics telemetry for Playbook Explanation, and this event should be removed.

File: ansible_wisdom/ai/api/utils/analytics_telemetry_model.py [hunk 2]
```diff
@@ -23,6 +23,8 @@ class AnalyticsTelemetryEvents(Enum):
     RECOMMENDATION_GENERATED = "Recommendation Generated"
     RECOMMENDATION_ACTION = "Recommendation Action"
     PRODUCT_FEEDBACK = "Product Feedback"
+    PLAYBOOK_GENERATION_ACTION = "Action on the Playbook Generation Wizard"
```
Comment (robinbobbitt) [inline ansible_wisdom/ai/api/utils/analytics_telemetry_model.py:26]: WDYT about naming this "Playbook Generation Action" for consistency? For the initial analytics telemetry, we consulted Segment's [naming convention recommendations](https://segment.com/academy/collecting-data/naming-conventions-for-clean-data/) and attempted to (roughly) follow their object-action framework.

File: ansible_wisdom/ai/api/utils/analytics_telemetry_model.py [hunk 3]
```diff
@@ -42,6 +44,31 @@ class AnalyticsRecommendationGenerated:
...
+class AnalyticsPlaybookExplanationGenerated:
+    explanation_id: str = field(validator=validators.instance_of(str), converter=str, default="")
+    rh_user_org_id: int = field(validator=validators.instance_of(int), converter=int, default=0)
+    timestamp: str = field(
+        default=Factory(lambda self: timezone.now().isoformat(), takes_self=True)
+    )
+
+
+@frozen
+class AnalyticsPlaybookGenerationWizard:
```
Comment (robinbobbitt) [inline ansible_wisdom/ai/api/utils/analytics_telemetry_model.py:57]: We need to add `model_name` to this event.

File: ansible_wisdom/ai/api/utils/seated_users_allow_list.py [hunk 4]
```diff
@@ -320,7 +321,26 @@
         "modelName": None,
         "rh_user_has_seat": None,
         "rh_user_org_id": None,
+        "response": None,
         "timestamp": None,
         "wizardId": None,
     },
+    "playbookGenerationAction": {
+        "action": None,
```
Comment (robinbobbitt) [inline ansible_wisdom/ai/api/utils/seated_users_allow_list.py:329]: We should add: 
```
        "modelName": None,
        "imageTags": None,
        "hostname": None,
        "groups": None,
        "rh_user_has_seat": None,
        "rh_user_org_id": None,
        "timestamp": None,
```

**Review by robinbobbitt — CHANGES_REQUESTED** [review]
Please see comments. Thanks!

File: tools/openapi-schema/ansible-ai-connect-service.yaml [hunk 5]
```diff
@@ -580,7 +580,7 @@ paths:
           description: One or more backend services are unavailable.
 components:
   schemas:
-    ActionEnum:
+    ActionDe0Enum:
```
Comment (robinbobbitt) [inline tools/openapi-schema/ansible-ai-connect-service.yaml:583]: Is this a typo?

Comment (goneri) [reply tools/openapi-schema/ansible-ai-connect-service.yaml:583]: I was also confused but it's from drf_spectacular. There is two kind of `ActionEnum` and it differentiate them like this.

Comment (robinbobbitt) [reply tools/openapi-schema/ansible-ai-connect-service.yaml:583]: Strange. I tried upgrading drf_spectacular to see if the "bug" was fixed and it just turned it to `Action4c6Enum`. :-P So I guess I'll ignore it! Thanks.

File: ansible_wisdom/ai/api/utils/analytics_telemetry_model.py [hunk 6]
```diff
@@ -42,6 +43,32 @@ class AnalyticsRecommendationGenerated:
     )
 
 
+@frozen
+class AnalyticsPlaybookExplanationGenerated:
```
Comment (robinbobbitt) [inline ansible_wisdom/ai/api/utils/analytics_telemetry_model.py:47]: Shall we delete this unused class?

File: ansible_wisdom/ai/api/views.py [hunk 7]
```diff
@@ -313,6 +324,37 @@ def write_to_segment(
...
+                "fromPage": from_page,
+                "toPage": to_page,
+                "modelName": model_name,
+            }
+            send_segment_event(event, "playbookGenerationAction", user)
+            if from_page > 1 and action in [1, 3]:
+                send_segment_analytics_event(
+                    AnalyticsTelemetryEvents.PLAYBOOK_GENERATION_ACTION,
+                    lambda: AnalyticsPlaybookGenerationWizard(
+                        action=action,
```
Comment (robinbobbitt) [inline ansible_wisdom/ai/api/views.py:351]: missing the model_name arg here.

**Review by robinbobbitt — APPROVED** [review]

**Review by robinbobbitt — CHANGES_REQUESTED** [review]
It looks like modelName is not getting set on explanation and generation. Please take a look. Thanks!

**Review by manstis — COMMENTED** [review]
Approving after rebase.

@robinbobbitt performed the _real_ review.

**Review by manstis — APPROVED** [review]
Approving after rebase.

@robinbobbitt performed the real review.

**Pull request merged**